lib/api/src/sys/externals/function.rs

```diff
@@ -983,8 +1098,30 @@ mod inner {
         fn call_trampoline_address() -> VMTrampoline;
     }
 
+    /// Empty trait to specify the kind of `HostFunction`: With or
+    /// without an environment.
+    ///
+    /// This trait is never aimed to be used by a user. It is used by
+    /// the trait system to automatically generate the appropriate
+    /// host functions.
+    #[doc(hidden)]
+    pub trait HostFunctionKind {}
```

Comment:
Traits that are not meant to be implemented outside the crate should use the [sealed trait pattern](https://rust-lang.github.io/api-guidelines/future-proofing.html#c-sealed) to ensure this on compile-time